### PR TITLE
Bluetooth: controller: LLCP: Fix pause encryption invalid behaviour

### PR DESCRIPTION
**Pull request by thoh-ot**
Fixes an issue regarding handling invalid pause encryption behaviour in the peripheral LLCP state machine.

### CODE REVIEW

File: tests/bluetooth/controller/mock_ctrl/src/ull_conn_iso.c [hunk 1]
```diff
@@ -42,21 +42,23 @@
 #include "hal/debug.h"
 
 static struct ll_conn_iso_group cig = { 0 };
-static struct ll_conn_iso_stream cis = { .established = 1};
+static struct ll_conn_iso_stream cis = { 0 };
 
-struct ll_conn_iso_stream *ll_conn_iso_stream_get_by_acl(struct ll_conn *conn, uint16_t *cis_iter)
+__attribute__((weak))
```
Comment (kruithofa) [inline tests/bluetooth/controller/mock_ctrl/src/ull_conn_iso.c:47]: in include/zephyr/toolchain/gcc.h __weak is defined as __attribute__((weak)), and this is used for weak functions
```suggestion
__weak
```

File: tests/bluetooth/controller/mock_ctrl/src/ull_conn_iso.c [hunk 2]
```diff
@@ -42,21 +42,23 @@
...
+struct ll_conn_iso_stream *ll_conn_iso_stream_get_by_acl(struct ll_conn *conn,
+							 uint16_t *cis_iter)
 {
-	return &cis;
+	return NULL;
 }
 
-struct ll_conn_iso_stream *ll_conn_iso_stream_get(uint16_t handle)
+__attribute__((weak)) struct ll_conn_iso_stream *ll_conn_iso_stream_get(uint16_t handle)
```
Comment (kruithofa) [inline tests/bluetooth/controller/mock_ctrl/src/ull_conn_iso.c:54]:
```suggestion
__weak struct ll_conn_iso_stream *ll_conn_iso_stream_get(uint16_t handle)
```

**Review by kruithofa — CHANGES_REQUESTED** [review]

File: tests/bluetooth/controller/mock_ctrl/src/ull_conn_iso.c (hunk 1 above)
Comment (thoh-ot) [reply tests/bluetooth/controller/mock_ctrl/src/ull_conn_iso.c]: OK

File: tests/bluetooth/controller/mock_ctrl/src/ull_conn_iso.c (hunk 2 above)
Comment (thoh-ot) [reply tests/bluetooth/controller/mock_ctrl/src/ull_conn_iso.c]: OK

File: tests/bluetooth/controller/ctrl_cis_create/src/main.c [hunk 3]
```diff
@@ -44,9 +48,14 @@
 
 static struct ll_conn conn;
 
+/* struct ll_conn_iso_stream *ll_conn_iso_stream_get(uint16_t handle); */
+FAKE_VALUE_FUNC(struct ll_conn_iso_stream *, ll_conn_iso_stream_get, uint16_t /*handle*/);
```
[the remark this answers is not in the document]
Comment (thoh-ot) [reply tests/bluetooth/controller/ctrl_cis_create/src/main.c:52]: OK

**Review by kruithofa — APPROVED** [review]
LGTM

**Review by erbr-ot — APPROVED** [review]
LGTM

**Review by cvinayak — APPROVED** [review]

**Pull request merged**